FilePersistence/src/version_control/Commit.cpp

```diff
@@ -118,7 +118,7 @@ bool Commit::isValidMatch(QString content, int indexOfId, int& start, int& end,
 	start++;
 
 	// end is the character after the line containing id
-	while (end <= content.size() && content[end] != '\n')
+	while (content[end] != '\0' && content[end] != '\n')
```

Comment:
this condition is not good, you need a separate parameter to the `isValidMatch` function that tells you what the size of the content is and you should check that here. You can get the size directly from the commit object.